api/v1alpha1/inferencepool_types.go

```diff
@@ -61,8 +61,85 @@ type InferencePoolSpec struct {
 	// +kubebuilder:validation:Maximum=65535
 	// +kubebuilder:validation:Required
 	TargetPortNumber int32 `json:"targetPortNumber"`
+
+	// EndpointPickerConfig selects and configures the endpoint picking algorithm to apply on the requests sent
+	// to this pool.
+	EndpointPickerConfig `json:"endpointPickerConfig"`
+}
+
+// EndpointPickerConfig selects and configures the endpoint picking algorithm to apply on the requests sent.
+// Only one of the following options can be set.
+type EndpointPickerConfig struct {
+	// Extension configures an endpoint picker as an extension service.
+	//
+	// +optional
+	Extension *ExtensionConfig `json:"extension"`
+	// Algorithm configures the endpoint picker by a name that the provider understands
+	// and knows how to set it up with its own gateway implementation.
+	//
+	// +optional
+	Algorithm *AlgorithmConfig `json:"algorithm"`
+}
+
+// AlgorithmConfig is a reference to the endpoint picking algorithm that the gateway should apply.
+type AlgorithmConfig struct {
+	// Name is an identifier of the algorithm that the gateway should employ.
+	Name string `json:"name"`
+}
+
+// ExtensionConfig specifies how to instantiate and configure an extension that implements the endpoint picking
+// algorithm.
+type ExtensionConfig struct {
+	// Specifies how the deployment for the extension gets configured.
+	ExtensionDeployment `json:"extensionDeployment"`
+	// Configurations for the connection between the LB and the extension.
+	ExtensionConnection `json:"extensionConnection"`
+}
+
+// ExtensionDeployment encapsulates the parameters for how to instantiate the extension deployment.
+type ExtensionDeployment struct {
+	// A reference to a deployed extension.
+	// <unresolved>
+	// Whether or not ExtensionRef is required is still not resolved.
+	// The debate is if we want to offer a path for providers to deploy managed extensions where
+	// if the ref is not set, then it is assumed that the gateway controller will create it.
+	// </unresolved>
+	ExtensionRef *ExtensionRef `json:"extensionRef"`
 }
 
+// ExtensionRef is a reference to the extension deployment.
+type ExtensionRef struct {
+	// A selector for the pods that run the deployment.
+	//
+	// +kubebuilder:validation:Required
+	Selector map[string]string `json:"selector"`
```

Comment:
@danehans @robscott If we change this to a Service ref, then the absence of it could mean that the extension is deployed as a sidecar.

Reply:
If this is a generic k8s object ref that defaults to Service, is there any way that we could reuse that to describe a sidecar? @danehans or @arkodg do you have any thoughts on what the ideal API surface would look like for an extension deployed as a sidecar?

Reply:
assuming sidecar here, is sidecar in the gateway pod, if it looks like a `backendRef`, you could route back to the Gateway Service and have the sidecar listen on a different port, and other implementations like Envoy Gateway could use the custom `Backend` Resource to `backendRef` to route via UDS

Reply:
Yep, sidecar on the gateway pod, that's a good idea.

Reply:
ok, so we should be good with a service reference here, right?

Reply:
@ahg-g something that looks like https://github.com/kubernetes-sigs/gateway-api/blob/50cf145b0dd1849ebd03c5866a92797862f23efe/apis/v1/object_reference_types.go#L96 should solve the `Service` case and also allow for other custom backend types

Reply:
> you could route back to the Gateway Service and have the sidecar listen on a different port.

This does work, but UDS is more performant than a network socket and should be supported.

What are your thoughts on exposing additional connection config for InferencePool, similar to Gateway listener? See this [comment](https://github.com/kubernetes-sigs/gateway-api-inference-extension/pull/162#discussion_r1910891232) for an example.